### ISSUE TITLE
Merge the all documentation in the /docs folder

### ISSUE THREAD

**Issue by jooola** [hetznercloud/hcloud-cloud-controller-manager#483]
The documentation is currently split between the README.md and the /docs folder.

To ensure users of this tool properly configure their setup, we need to improve the documentation so that users know where to find information and don't miss important information when configuring HCCM.

I propose to merge all the documentation in the /docs folder, provide a good documentation entry point that helps to navigate through it, and add some missing and critical documentation.

**Comment by lukasmetzner** [comment]
What would should be kept in the README.md?
- Intro
- Features
- Versioning policy (?)
- Development (?)
- License

**Comment by jooola** [comment]
In my opinion: 

 - Intro
- Versioning policy ⇒ Maybe renamed to `Releases` and add a link to the changelog?
- Development
  - If this sections grows too large, keep the title, but move the content in the docs/ 
- License

**Comment by lukasmetzner** [comment]
Removing the [Features](https://github.com/hetznercloud/hcloud-cloud-controller-manager?tab=readme-ov-file#features) section, or would you account it to the intro?

**Comment by jooola** [comment]
Yes, I first meant to move the feature section in the docs, but maybe merging it in the intro is a better idea. But having a "features" section seem unnecessary.

**Comment by lukasmetzner** [comment]
Is moving to a hosted (html rendered) docs solution an option? mdbook, etc.

**Comment by lukasmetzner** [comment]
> Is moving to a hosted (html rendered) docs solution an option? mdbook, etc.

Had a small discussion about this topic with @jooola. Probably best to start with Markdown docs for now and handle the topic of serving in a parallel stream.

**Comment by jooola** [comment]
This third party project already wrote a decent amount of docs, maybe it is worth getting some inspiration from it:
- https://github.com/vitobotta/hetzner-k3s/tree/main/docs
